code/modules/surgery/bodyparts/helpers.dm

```diff
@@ -137,29 +137,25 @@
 			disabled += zone
 	return disabled
 
-//Remove all embedded objects from all limbs on the carbon mob
-/mob/living/carbon/proc/remove_all_embedded_objects()
-	var/turf/T = get_turf(src)
+///Remove a specific embedded item from the carbon mob
+/mob/living/carbon/proc/remove_embedded_object(obj/item/I)
+	SEND_SIGNAL(src, COMSIG_HUMAN_EMBED_REMOVAL, I)
```

Comment:
In general we don't want signals that are disguised procs.
If you are calling `remove_embedded_object()` on many places you can make several signals, so we can better track their sources.